### PR TITLE
Move mapping classes into a separate package

### PR DESCRIPTION
**Pull request by stevenschlansker**
InferredMapperFactory -> InferredRowMapperFactory
Move RegisteredRowMapper to a lambda on SqlStatement

(requires #385)

### CODE REVIEW

File: core/src/main/java/org/jdbi/v3/mapper/InferredRowMapperFactory.java [hunk 1]
```diff
@@ -11,21 +11,22 @@
...
 import static org.jdbi.v3.Types.findGenericParameter;
 
 import java.lang.reflect.Type;
 import java.util.Optional;
 
+import org.jdbi.v3.StatementContext;
 import org.jdbi.v3.tweak.RowMapper;
 
-class InferredMapperFactory implements RowMapperFactory
+public class InferredRowMapperFactory implements RowMapperFactory
```
Comment (qualidafial) [inline core/src/main/java/org/jdbi/v3/mapper/InferredRowMapperFactory.java]: If this class is going public, we probably want to add a short javadoc blurb explaining what it's for.

**Comment by qualidafial** [comment]
One minor comment to consider, then :shipit:

**Comment by qualidafial** [comment]
Move `MappingRegistry` too?

**Comment by stevenschlansker** [comment]
I wasn't planning to move `MappingRegistry` or `ArgumentRegistry` since we would have to make them public, and they are fairly tied to `JdbiConfig`.

**Pull request merged**